tests/bluetooth/audio/mocks/include/gatt_expects.h

```diff
@@ -0,0 +1,111 @@
+												   \
+	IF_NOT_EMPTY(_attr, (                                                                      \
+		     zassert_equal_ptr(_attr, bt_gatt_attr_read_fake.arg1_val,                     \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "attr");))                                       \
+												   \
+	IF_NOT_EMPTY(_buf, (                                                                       \
+		     zassert_equal_ptr(_buf, bt_gatt_attr_read_fake.arg2_val,                      \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "buf");))                                        \
+												   \
+	IF_NOT_EMPTY(_buf_len, (                                                                   \
+		     zassert_equal(_buf_len, bt_gatt_attr_read_fake.arg3_val,                      \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "_buf_len");))                                   \
+												   \
+	IF_NOT_EMPTY(_offset, (                                                                    \
+		     zassert_equal(_offset, bt_gatt_attr_read_fake.arg4_val,                       \
+				   "'%s()' was called with incorrect '%s' value",                  \
+				   func_name, "offset");))                                         \
+												   \
+	/* assert if _data is valid, but _len is empty */                                          \
+	IF_EMPTY(_value_len, (IF_NOT_EMPTY(_value, (zassert_unreachable();))))                     \
+												   \
+	IF_NOT_EMPTY(_value_len, (                                                                 \
+		     zassert_equal(_value_len, bt_gatt_attr_read_fake.arg6_val,                    \
+				   "'%s()' was called with incorrect '%s' value",                  \
+				   func_name, "value_len");                                        \
+		     expect_data(func_name, "value", _value, bt_gatt_attr_read_fake.arg5_val,      \
+				 _value_len);))                                                    \
+} while (0)
+
+#define expect_bt_gatt_notify_cb_called_once(_conn, _uuid, _attr, _data, _len)                     \
+do {                                                                                               \
+	const char *func_name = "bt_gatt_notify_cb";                                               \
+	struct bt_gatt_notify_params *params;                                                      \
+												   \
+	IF_NOT_EMPTY(_conn, (                                                                      \
+		     zassert_equal_ptr(_conn, mock_bt_gatt_notify_cb_fake.arg0_val,                \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "conn");))                                       \
+												   \
+	params = mock_bt_gatt_notify_cb_fake.arg1_val;                                             \
```

Comment:
Do we need a IF_NOT_EMPTY check for `arg1_val` before assigning it to `params` and using `params` right after?

Reply:
No, as it's ensured to be valid pointer. `params` are stack allocated, thus the `bt_gatt_notify_cb` mock function does deep copy of it and ensures the `bt_gatt_notify_cb` was called with valid parameters:
audio/mocks/src/gatt.c
```
	zassert_not_null(params, "'%s()' was called with incorrect '%s' value", __func__, "params");

	/* Either params->uuid, params->attr, or both has to be provided */
	zassert_true(params->uuid != NULL || params->attr != NULL,
		     "'%s()' was called with incorrect '%s' value", __func__,
		     "params->uuid or params->attr");
```